### ISSUE TITLE
RFE: Support for Azure Gov OMS endpoints

### ISSUE THREAD

**Issue by chen23** [F5Networks/f5-telemetry-streaming#20]
##### ISSUE TYPE

 - Feature Idea

##### SUMMARY

Support Azure Gov OMS
https://docs.microsoft.com/en-us/azure/azure-government/documentation-government-services-monitoringandmanagement

**Comment by dstokesf5** [comment]
I have added this issue to our internal product backlog as AT-768. Thank you for your feedback.

**Comment by Mikej81** [comment]
Just verified that this is still not working.

[workspaceId].ods.opinsights.azure.com | [workspaceId].ods.opinsights.azure.us

`[xadmin@bigip0:Active:In Sync] ~ # tail -f /var/log/restnoded/restnoded.log
Thu, 26 Mar 2020 17:09:20 GMT - severe: [telemetry.Azure_Log_Analytics] error: getaddrinfo ENOTFOUND 28bbaaad-089d-4d46-af97-cf480a9fa598.ods.opinsights.azure.com 28bbaaad-089d-4d46-af97-cf480a9fa598.ods.opinsights.azure.com:443
Thu, 26 Mar 2020 17:10:20 GMT - severe: [telemetry.Azure_Log_Analytics] error: getaddrinfo ENOTFOUND 28bbaaad-089d-4d46-af97-cf480a9fa598.ods.opinsights.azure.com 28bbaaad-089d-4d46-af97-cf480a9fa598.ods.opinsights.azure.com:443
Thu, 26 Mar 2020 17:11:20 GMT - severe: [telemetry.Azure_Log_Analytics] error: getaddrinfo ENOTFOUND 28bbaaad-089d-4d46-af97-cf480a9fa598.ods.opinsights.azure.com 28bbaaad-089d-4d46-af97-cf480a9fa598.ods.opinsights.azure.com:443
Thu, 26 Mar 2020 17:12:20 GMT - severe: [telemetry.Azure_Log_Analytics] error: getaddrinfo ENOTFOUND 28bbaaad-089d-4d46-af97-cf480a9fa598.ods.opinsights.azure.com 28bbaaad-089d-4d46-af97-cf480a9fa598.ods.opinsights.azure.com:443
Thu, 26 Mar 2020 17:13:20 GMT - severe: [telemetry.Azure_Log_Analytics] error: getaddrinfo ENOTFOUND 28bbaaad-089d-4d46-af97-cf480a9fa598.ods.opinsights.azure.com 28bbaaad-089d-4d46-af97-cf480a9fa598.ods.opinsights.azure.com:443
Thu, 26 Mar 2020 17:14:20 GMT - severe: [telemetry.Azure_Log_Analytics] error: getaddrinfo ENOTFOUND 28bbaaad-089d-4d46-af97-cf480a9fa598.ods.opinsights.azure.com 28bbaaad-089d-4d46-af97-cf480a9fa598.ods.opinsights.azure.com:443
Thu, 26 Mar 2020 17:15:20 GMT - severe: [telemetry.Azure_Log_Analytics] error: getaddrinfo ENOTFOUND 28bbaaad-089d-4d46-af97-cf480a9fa598.ods.opinsights.azure.com 28bbaaad-089d-4d46-af97-cf480a9fa598.ods.opinsights.azure.com:443
Thu, 26 Mar 2020 17:16:20 GMT - severe: [telemetry.Azure_Log_Analytics] error: getaddrinfo ENOTFOUND 28bbaaad-089d-4d46-af97-cf480a9fa598.ods.opinsights.azure.com 28bbaaad-089d-4d46-af97-cf480a9fa598.ods.opinsights.azure.com:443
Thu, 26 Mar 2020 17:17:20 GMT - severe: [telemetry.Azure_Log_Analytics] error: getaddrinfo ENOTFOUND 28bbaaad-089d-4d46-af97-cf480a9fa598.ods.opinsights.azure.com 28bbaaad-089d-4d46-af97-cf480a9fa598.ods.opinsights.azure.com:443
Thu, 26 Mar 2020 17:18:20 GMT - severe: [telemetry.Azure_Log_Analytics] error: getaddrinfo ENOTFOUND 28bbaaad-089d-4d46-af97-cf480a9fa598.ods.opinsights.azure.com 28bbaaad-089d-4d46-af97-cf480a9fa598.ods.opinsights.azure.com:443
Thu, 26 Mar 2020 17:19:20 GMT - severe: [telemetry.Azure_Log_Analytics] error: getaddrinfo ENOTFOUND 28bbaaad-089d-4d46-af97-cf480a9fa598.ods.opinsights.azure.com 28bbaaad-089d-4d46-af97-cf480a9fa598.ods.opinsights.azure.com:443
Thu, 26 Mar 2020 17:20:20 GMT - severe: [telemetry.Azure_Log_Analytics] error: getaddrinfo ENOTFOUND 28bbaaad-089d-4d46-af97-cf480a9fa598.ods.opinsights.azure.com 28bbaaad-089d-4d46-af97-cf480a9fa598.ods.opinsights.azure.com:443
Thu, 26 Mar 2020 17:21:20 GMT - severe: [telemetry.Azure_Log_Analytics] error: getaddrinfo ENOTFOUND 28bbaaad-089d-4d46-af97-cf480a9fa598.ods.opinsights.azure.com 28bbaaad-089d-4d46-af97-cf480a9fa598.ods.opinsights.azure.com:443
Thu, 26 Mar 2020 17:22:20 GMT - severe: [telemetry.Azure_Log_Analytics] error: getaddrinfo ENOTFOUND 28bbaaad-089d-4d46-af97-cf480a9fa598.ods.opinsights.azure.com 28bbaaad-089d-4d46-af97-cf480a9fa598.ods.opinsights.azure.com:443
Thu, 26 Mar 2020 17:23:20 GMT - severe: [telemetry.Azure_Log_Analytics] error: getaddrinfo ENOTFOUND 28bbaaad-089d-4d46-af97-cf480a9fa598.ods.opinsights.azure.com 28bbaaad-089d-4d46-af97-cf480a9fa598.ods.opinsights.azure.com:443
Thu, 26 Mar 2020 17:24:20 GMT - severe: [telemetry.Azure_Log_Analytics] error: getaddrinfo ENOTFOUND 28bbaaad-089d-4d46-af97-cf480a9fa598.ods.opinsights.azure.com 28bbaaad-089d-4d46-af97-cf480a9fa598.ods.opinsights.azure.com:443
Thu, 26 Mar 2020 17:25:20 GMT - severe: [telemetry.Azure_Log_Analytics] error: getaddrinfo ENOTFOUND 28bbaaad-089d-4d46-af97-cf480a9fa598.ods.opinsights.azure.com 28bbaaad-089d-4d46-af97-cf480a9fa598.ods.opinsights.azure.com:443
Thu, 26 Mar 2020 17:26:20 GMT - severe: [telemetry.Azure_Log_Analytics] error: getaddrinfo ENOTFOUND 28bbaaad-089d-4d46-af97-cf480a9fa598.ods.opinsights.azure.com 28bbaaad-089d-4d46-af97-cf480a9fa598.ods.opinsights.azure.com:443
Thu, 26 Mar 2020 17:27:20 GMT - severe: [telemetry.Azure_Log_Analytics] error: getaddrinfo ENOTFOUND 28bbaaad-089d-4d46-af97-cf480a9fa598.ods.opinsights.azure.com 28bbaaad-089d-4d46-af97-cf480a9fa598.ods.opinsights.azure.com:443
Thu, 26 Mar 2020 17:28:20 GMT - severe: [telemetry.Azure_Log_Analytics] error: getaddrinfo ENOTFOUND 28bbaaad-089d-4d46-af97-cf480a9fa598.ods.opinsights.azure.com 28bbaaad-089d-4d46-af97-cf480a9fa598.ods.opinsights.azure.com:443
Thu, 26 Mar 2020 17:29:20 GMT - severe: [telemetry.Azure_Log_Analytics] error: getaddrinfo ENOTFOUND 28bbaaad-089d-4d46-af97-cf480a9fa598.ods.opinsights.azure.com 28bbaaad-089d-4d46-af97-cf480a9fa598.ods.opinsights.azure.com:443
`

**Comment by Mikej81** [comment]
Seems like this could just be resolved by checking for the region in the VM Metadata Service.
`[xadmin@bigip0:Active:In Sync] ~ # curl -H Metadata:true "http://169.254.169.254/metadata/instance?api-version=2017-08-01" --interface mgmt`

```json
{
    "compute":{
        "location":"usgovvirginia",
        "name":"f5dns99-bigip0",
        "offer":"f5-big-ip-best",
        "osType":"Linux",
        "placementGroupId":"",
        "platformFaultDomain":"0",
        "platformUpdateDomain":"0",
        "publisher":"f5-networks",
        "resourceGroupName":"mjc_1t_payg_lab",
        "sku":"f5-bigip-virtual-edition-1g-best-hourly",
        "subscriptionId":"a0b713be-1237-4769-8647-f0f281a998c9",
        "tags":"application:APP;cost:COST;environment:ENV;group:GROUP;owner:OWNER",
        "version":"14.1.200000",
        "vmId":"84f1f13f-35f8-472d-bea1-c1ee11bd1e71",
        "vmSize":"Standard_DS4_v2"
        },
        "network":{
            "interface":[
                {"ipv4":{"ipAddress":[
                    {"privateIpAddress":"192.168.1.4","publicIpAddress":""}
                    ],
                    "subnet":[{"address":"192.168.1.0","prefix":"24"}]},"ipv6":{"ipAddress":[]},"macAddress":"001DD800A40F"},{"ipv4":{"ipAddress":[{"privateIpAddress":"192.168.3.4","publicIpAddress":""},{"privateIpAddress":"192.168.3.14","publicIpAddress":""}],"subnet":[{"address":"192.168.3.0","prefix":"24"}]},"ipv6":{"ipAddress":[]},"macAddress":"001DD800A1A6"},{"ipv4":{"ipAddress":[{"privateIpAddress":"192.168.2.4","publicIpAddress":""},{"privateIpAddress":"192.168.2.10","publicIpAddress":""}],"subnet":[{"address":"192.168.2.0","prefix":"24"}]},"ipv6":{"ipAddress":[]},"macAddress":"001DD800A943"}]}}
```

I forked the repo to tweak a fed version to be a gap fix.

**Comment by Mikej81** [comment]
Here you go, mocked up a solution!  

[https://github.com/F5Networks/f5-telemetry-streaming/compare/master...Mikej81:master](https://github.com/F5Networks/f5-telemetry-streaming/compare/master...Mikej81:master)

**Comment by Mikej81** [comment]
Attempted to dig into providing region with context as part of the TS declaration, but dont think that would help other than which logic to evaluate, but it could remove the VM Metadata call.

[https://github.com/F5Networks/f5-telemetry-streaming/blob/master/src/schema/latest/consumer_schema.json#L101](https://github.com/F5Networks/f5-telemetry-streaming/blob/master/src/schema/latest/consumer_schema.json#L101)

```json
{
    "class": "Telemetry",
    "My_Consumer": {
        "class": "Telemetry_Consumer",
        "type": "Azure_Log_Analytics",
        "region": "usgovvirginia"
        "workspaceId": "workspaceid",
        "passphrase": {
            "cipherText": "sharedkey"
        }
    }
}
```
-
```json
[
  {
    "displayName": "USGov Virginia",
    "id": "/subscriptions/a0b713be-1237-4769-8647-f0f281a998c9/locations/usgovvirginia",
    "latitude": "37.3719",
    "longitude": "-79.8164",
    "name": "usgovvirginia",
    "subscriptionId": null
  },
  {
    "displayName": "USGov Iowa",
    "id": "/subscriptions/a0b713be-1237-4769-8647-f0f281a998c9/locations/usgoviowa",
    "latitude": "41.5908",
    "longitude": "-93.6208",
    "name": "usgoviowa",
    "subscriptionId": null
  },
  {
    "displayName": "USDoD East",
    "id": "/subscriptions/a0b713be-1237-4769-8647-f0f281a998c9/locations/usdodeast",
    "latitude": "36.6676",
    "longitude": "-78.3875",
    "name": "usdodeast",
    "subscriptionId": null
  },
  {
    "displayName": "USDoD Central",
    "id": "/subscriptions/a0b713be-1237-4769-8647-f0f281a998c9/locations/usdodcentral",
    "latitude": "41.6005",
    "longitude": "-93.6091",
    "name": "usdodcentral",
    "subscriptionId": null
  },
  {
    "displayName": "USGov Texas",
    "id": "/subscriptions/a0b713be-1237-4769-8647-f0f281a998c9/locations/usgovtexas",
    "latitude": "29.4241",
    "longitude": "-98.4936",
    "name": "usgovtexas",
    "subscriptionId": null
  },
  {
    "displayName": "USGov Arizona",
    "id": "/subscriptions/a0b713be-1237-4769-8647-f0f281a998c9/locations/usgovarizona",
    "latitude": "33.4484",
    "longitude": "-112.0740",
    "name": "usgovarizona",
    "subscriptionId": null
  }
]
```

**Comment by dstokesf5** [comment]
This was resolved as part of the 1.11.0 release.
